packages/application/src/submission.tsx

```diff
@@ -14,35 +14,44 @@
  * limitations under the License.
  */
 
+import '../style/index.css';
+
 import { showDialog, Dialog } from '@jupyterlab/apputils';
 import { URLExt } from '@jupyterlab/coreutils';
 import { ServerConnection } from '@jupyterlab/services';
 
 import * as React from 'react';
 
+const MESSAGE_DISPLAY = 'elyra-pipelineSubmission-messageDisplay';
+const ERROR_DISPLAY_BUTTON = 'elyra-pipelineSubmission-errDisplayButton';
+const DOWN_ICON_CLASS = 'elyra-pipelineSubmission-errDisplayButton-down';
+const UP_ICON_CLASS = 'elyra-pipelineSubmission-errDisplayButton-up';
+const ERROR_DETAILS_VISIBLE = 'elyra-pipelineSubmission-error-visible';
+const ERROR_DETAILS_HIDDEN = 'elyra-pipelineSubmission-error-hidden';
+
 export class SubmissionHandler {
   static handleError(
     response: any,
     submissionType: string
   ): Promise<Dialog.IResult<any>> {
-    let res_body = response['message'] ? response['message'] : '';
-    res_body = response['reason']
-      ? res_body + ': ' + response['reason']
-      : res_body;
-
+    const reason = response.reason ? response.reason : '';
+    const message = response.message ? response.message : '';
+    const traceback = response.traceback ? response.traceback : '';
     const default_body = 'Check the JupyterLab log for more details.';
 
     return showDialog({
       title: 'Error submitting ' + submissionType,
-      body: res_body ? (
-        <p>
-          {res_body}
-          <br />
-          {default_body}
-        </p>
-      ) : (
-        <p>{default_body}</p>
-      ),
+      body:
```

Comment:
General question.  Is the `SubmissionHandler` used for all requests to the server?   I believe there are interactions when retrieving the metadata and submitting a notebook.  Will those go through the `SubmissionHandler` or be able to use `ErrorDialogContent`?

Reply:
Yes, it also applies to errors in submitting a notebook. The dialog box title would be 'Error submitting notebook', then we'd get the same ErrorDialogContent rendered below.